### PR TITLE
Add MLPerf logging + amendments

### PR DESCRIPTION
**Pull request by janekl**
I'm adding MLPerf logging in this PR + several minor changes.

I made a few the decisions including:
* removing test set from evaluation - because it's not used in MLPerf
* skipping epoch end evaluation - this is done within epoch anyway (the last evaluation step during an epoch).

This is to make training loop and logging adequate for MLPerf purposes.

Thanks to @mmarcinkiewicz for helping out with all this.

### CODE REVIEW

File: torchrec_dlrm/dlrm_main.py [hunk 1]
```diff
@@ -428,32 +465,49 @@ def _train(
...
+                        )
+                    is_success = True
+                    break
                 train_pipeline._model.train()
-        except StopIteration:
-            if is_rank_zero:
-                print("Total number of iterations:", it - 1)
-            break
+    except StopIteration:
+        # train_pipeline is exhausted
```
[the remark this answers is not in the document]
Comment (janekl) [reply torchrec_dlrm/dlrm_main.py:493]: Right, this is exactly what I meant too.

I'll comment perhaps that "train_pipeline completed"?

File: torchrec_dlrm/dlrm_main.py [hunk 2]
```diff
@@ -483,25 +543,34 @@ def train_val_test(
             lr_scheduler,
             args.print_lr,
             args.validation_freq_within_epoch,
+            args.validation_auroc,
             args.limit_train_batches,
             args.limit_val_batches,
+            not args.print_progress,
         )
-        val_auroc = _evaluate(
```
[the remark this answers is not in the document]
Comment (janekl) [reply torchrec_dlrm/dlrm_main.py:489]: Yes, you're right - but we don't need this evaluation step in MLPerf. Besides, we're going to set `--validation_freq_within_epoch $((TOTAL_TRAINING_SAMPLES / (GLOBAL_BATCH_SIZE * NUM_EVAL)))` so the last inter-validation call should correspond to the validation score reported in the line removed here (modulo a few remaining batches). This is how evaluation should be performed in the case of MLPerf.

File: torchrec_dlrm/dlrm_main.py [hunk 3]
```diff
@@ -243,12 +229,11 @@ def parse_args(argv: List[str]) -> argparse.Namespace:
         default=None,
         help="Frequency at which validation will be run within an epoch.",
     )
-    parser.set_defaults(
```
[the remark this answers is not in the document]
Comment (janekl) [reply torchrec_dlrm/dlrm_main.py:246]: Each of these CLI arguments in its definition sets `action="store_true"` and should default to `False` which is equivalent in boolean context to using `None`. I think there is no use in `set_defaults` here that's why I removed it.

File: torchrec_dlrm/dlrm_main.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (samiwilf) [reply torchrec_dlrm/dlrm_main.py:489]: The last intra-epoch validation AUROC won't be the same AUROC as when run after the first epoch.  TOTAL_TRAINING_SAMPLES  is not evenly divisible by (GLOBAL_BATCH_SIZE * NUM_EVAL)

File: torchrec_dlrm/dlrm_main.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (samiwilf) [reply torchrec_dlrm/dlrm_main.py]: "Dataset traversal complete" ?

File: torchrec_dlrm/dlrm_main.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (janekl) [reply torchrec_dlrm/dlrm_main.py:489]: Right, I see your point on the difference between evaluation score for `epoch_num=0.99...` and `epoch_num=1.0`, I observed that when doing convergence studies. In any case this is how DLRM benchmark worked in MLPerf <= v2.1 (see [here](https://github.com/mlcommons/training_results_v2.1/blob/main/NVIDIA/benchmarks/dlrm/implementations/hugectr-preview/dgx_h100.py#L249) and [here](https://github.com/facebookresearch/dlrm/blob/mlperf/dlrm_s_pytorch.py#L575)) so I think we can stick to that and set `--validation_freq_within_epoch` appropriately to get virtually the same evaluation score in the last intra-epoch validation and post-epoch validation.

I would say that's fine and it is "backward compatible" with how DLRM worked in past.

**Comment by janekl** [comment]
MLPerf logging was integrated in the official reference implementation in https://github.com/mlcommons/training/pull/616. I'm closing the PR initially filed here.
